lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -99,8 +108,15 @@ class _Visitor extends SimpleAstVisitor {
     if (node is FunctionExpressionInvocation) {
       restOfElements = _extractElementsOfSimpleIdentifiers(node.function);
     } else if (node is MethodInvocation && node.target != null) {
+      if (_hasQuestionPeriod(node) ||
+          _hasQuestionPeriod(node.target) ||
+          DartTypeUtilities
+              .traverseNodesInDFS(node.target)
+              .any(_hasQuestionPeriod)) {
+        return;
```

Comment:
I think this will catch more than you intend. For example: `(p) a[b?.m1()].m2(p)`. I think it's semantically equivalent to replace this with a tear-off of `a[b?.m1()].m2` (though not necessarily easier to read, so maybe it's better that we don't lint in this case).

If you do want to create a lint in the above case, then I think you want to recursively look at the target of any `PropertyAccess`, `IndexExpression`, etc.